src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,206 @@
+        : null;
+
+      return {
+        ...person,
+        ...(findFather && { father: findFather }),
+        ...(findMother && { mother: findMother }),
+      };
+    });
+
+    return peopleList;
+  };
+
+  const filter = (params: URLSearchParams) => {
+    const sex = params.get('sex');
+    const query = params.get('query');
+    const centuries = params.getAll('centuries');
+
+    setFilteredList(people);
+
+    if (sex) {
+      setFilteredList((current) => {
+        return current?.filter((person) => person.sex === sex);
+      });
+    }
+
+    if (query) {
+      setFilteredList((current) => current?.filter((person) => {
+        const searchQuery = query.toLocaleLowerCase();
+        const name = person.name.toLocaleLowerCase();
+        const motherName = String(person.motherName).toLocaleLowerCase();
+        const fatherName = String(person.fatherName).toLocaleLowerCase();
+
+        return (
+          name.includes(searchQuery)
+            || motherName.includes(searchQuery)
+            || fatherName.includes(searchQuery)
+        );
+      }));
+    }
+
+    if (centuries.length) {
+      setFilteredList((current) => current?.filter((person) => {
+        const personCentury = Math.ceil(person.born / 100);
+
+        return centuries.includes(String(personCentury));
+      }));
+    }
+  };
+
+  const sort = (params: URLSearchParams) => {
+    const sortBy = params.get('sort');
+    const orderBy = params.get('order');
+    const sortByText = sortBy === 'name' || sortBy === 'sex';
+
+    if (sortBy && orderBy === 'desc') {
+      setFilteredList((current) => current?.sort((a: Person, b: Person) => {
+        if (sortByText) {
+          return a[sortBy].localeCompare(b[sortBy]);
+        }
+
+        if (sortBy === 'born' || sortBy === 'died') {
+          return a[sortBy] - b[sortBy];
+        }
+
+        return 1;
+      }));
+    }
+
+    if (sortBy && !orderBy) {
+      setFilteredList((current) => current?.sort((a: Person, b: Person) => {
+        if (sortByText) {
+          return b[sortBy].localeCompare(a[sortBy]);
+        }
+
+        if (sortBy === 'born' || sortBy === 'died') {
+          return b[sortBy] - a[sortBy];
+        }
+
+        return 1;
+      }));
+    }
+  };
+
+  useEffect(() => {
+    getPeople()
+      .then((response) => response)
+      .then((result) => {
+        const improvedList = addMotherAndFather(result);
+
+        setPeople(improvedList);
+        setFilteredList(improvedList);
+
+        if (result.length) {
+          setPageState('showTable');
+        } else {
+          setPageState('nothingFound');
+        }
+      })
+      .catch(() => setPageState('error'));
+  }, []);
+
+  useEffect(() => {
+    filter(searchParams);
+    sort(searchParams);
+  }, [people]);
+
+  useEffect(() => {
+    setUrlSlug(slug);
+  }, [slug]);
+
+  useEffect(() => {
+    filter(searchParams);
+    sort(searchParams);
+  }, [searchParams]);
```

Comment:
why not join the two? 🤔 
```suggestion
  useEffect(() => {
    filter(searchParams);
    sort(searchParams);
  }, [people, searchParams]);

  useEffect(() => {
    setUrlSlug(slug);
  }, [slug]);
```